src/EditorFeatures/Core/Shared/Tagging/EventSources/TaggerEventSources.SemanticChangedEventSource.cs

```diff
@@ -48,7 +48,10 @@ private void OnSubjectBufferChanged(object sender, TextContentChangedEventArgs e
             {
                 // Whenever this subject buffer has changed, we always consider that to be a 
                 // semantic change.
-                this.RaiseChanged();
+                if (e.Changes.Any())
```

Comment:
For the first question, do you get a `TextChanged` event?  If we want to deal with that, shouldn't we be listening to `ITextBuffer.ContentTypeChanged` instead?

For the second question, `OnOpenedDocumentSemanticChanged` goes out of it's way not to raise changes for itself *because* it assumes this method already has.